datafusion/sqllogictest/test_files/scalar.slt

```diff
@@ -1907,8 +1907,10 @@ select position('' in '')
 1
 
 
-query error POSITION function can only accept strings
+query I
 select position(1 in 1)
+----
+1
```

Comment:
This query used to fail also before https://github.com/apache/datafusion/commit/1b3608da7ca59d8d987804834d004e8b3e349d18 , i don't know why.

According to https://github.com/apache/datafusion/blob/8db30e25d6fe65a9779d237cf48aea9aee297502/datafusion/expr/src/type_coercion/functions.rs#L681-L682, every type is implicitly coercible into a Utf8

Reply:
Isn't this function only accept string? Why do we return 1 now